Spigot-API-Patches/0005-Adventure.patch

```diff
@@ -399,8 +399,80 @@ index 0000000000000000000000000000000000000000..13c5df5fb8ce1d0203d99e88dd691019
 +        return HANDLERS;
 +    }
 +}
+diff --git a/src/main/java/io/papermc/paper/text/PaperComponents.java b/src/main/java/io/papermc/paper/text/PaperComponents.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..643895529dcd17d3b7bc3de78835ca3fae07ee3d
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/text/PaperComponents.java
+@@ -0,0 +1,66 @@
++package io.papermc.paper.text;
++
++import net.kyori.adventure.text.Component;
++import net.kyori.adventure.text.flattener.ComponentFlattener;
++import net.kyori.adventure.text.format.NamedTextColor;
++import net.kyori.adventure.text.serializer.gson.GsonComponentSerializer;
++import net.kyori.adventure.text.serializer.plain.PlainComponentSerializer;
++import org.bukkit.Bukkit;
++import org.checkerframework.checker.nullness.qual.NonNull;
++
++/**
++ * Paper API-specific methods for working with {@link Component}s and related.
++ */
++public interface PaperComponents {
++    /**
++     * Get the {@link PaperComponents} instance.
++     *
++     * @return PaperComponents
++     */
++    static @NonNull PaperComponents paperComponents() {
++        return Bukkit.getPaperComponents();
++    }
++
++    /**
++     * Return a component flattener that can use game data to resolve extra information about components.
++     *
++     * @return a component flattener
++     */
++    @NonNull ComponentFlattener flattener();
++
++    /**
++     * Get a serializer for {@link Component}s that will convert components to
++     * a plain-text string.
++     *
++     * <p>Implementations may provide a serializer capable of processing any
++     * information that requires access to implementation details.</p>
++     *
++     * @return a serializer to plain text
++     */
++    @NonNull PlainComponentSerializer plainSerializer();
++
++    /**
++     * Get a serializer for {@link Component}s that will convert to and from the
++     * standard JSON serialization format using Gson.
++     *
++     * <p>Implementations may provide a serializer capable of processing any
++     * information that requires implementation details, such as legacy
++     * (pre-1.16) hover events.</p>
++     *
++     * @return a json component serializer
++     */
++    @NonNull GsonComponentSerializer gsonSerializer();
++
++    /**
++     * Get a serializer for {@link Component}s that will convert to and from the
++     * standard JSON serialization format using Gson, downsampling any RGB colors
++     * to their nearest {@link NamedTextColor} counterpart.
++     *
++     * <p>Implementations may provide a serializer capable of processing any
++     * information that requires implementation details, such as legacy
++     * (pre-1.16) hover events.</p>
++     *
++     * @return a json component serializer
++     */
++    @NonNull GsonComponentSerializer colorDownsamplingGsonSerializer();
```

Comment:
These should be made `static`, and have a separate ´Providers` interface for these non-`static` ones.